Homework_10_Java_Software_Messenger_Development/Messenger/src/User.java

```diff
@@ -0,0 +1,79 @@
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Class representing a user in the messenger system.
+ */
+
+public class User implements UserInterface {
+
+	private String name;
+	private Set<User> contacts = new HashSet<>();
```

Comment:
Using a more common type `Set` to handle a reference to a value of an inheritor `HashSet` is a good approach! 👍 

(this gives us flexibility when in the future the Collection type can be changed to any other inherited type if needed without breaking the functionality)